### PR TITLE
Fix cc_phone_home requiring 'tries' (SC-1101)

### PR DESCRIPTION
**Pull request by TheRealFalcon**
## Proposed Commit Message

```
Fix cc_phone_home requiring 'tries'

As part of dee8231, the code was updated to use a ValueError rather than
catching the top level Exception. This should have also included
catching TypeError. Also, calculate 'retries' correctly.

Additionally, add unit tests for module as it effectively had none.

LP: #1977952
```

I officially apologize for all of the younger-me rants about checked exceptions in other languages.

### CODE REVIEW

File: cloudinit/config/cc_phone_home.py [hunk 1]
```diff
@@ -190,7 +191,7 @@ def handle(name, cfg, cloud, log, args):
         url_helper.read_file_or_url(
             url,
             data=real_submit_keys,
-            retries=tries,
+            retries=tries - 1,
```
Comment (blackboxsw) [inline cloudinit/config/cc_phone_home.py:194]: +1 on the logic correction here. While this does represent change in behavior, I'd defend that original logic was a bug. Generally this retry logic will only trigger in significant repeated error conditions.

File: cloudinit/config/cc_phone_home.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ def handle(name, cfg, cloud, log, args):
     tries = ph_cfg.get("tries")
     try:
         tries = int(tries)  # pyright: ignore
-    except ValueError:
+    except (ValueError, TypeError):
```
Comment (blackboxsw) [inline cloudinit/config/cc_phone_home.py:133]: works for me. This is a better suggestion than the alternative of `tries = ph_cfg.get("tries", DEFAULT_RETRIES)`, as someone in the off chance provided `tries: null` in user-data, we'd still fail with the TypeError exception
